x-pack/plugins/security_solution/public/common/utils/download_blob.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/**
+ * Method for downloading any file
+ *
+ * @param blob raw data
+ * @param filename of file to be downloaded
+ *
+ */
+export const downloadBlob = (blob: Blob, filename: string) => {
+  const objectURL = window.URL.createObjectURL(blob);
+  const anchor = document.createElement('a');
+  anchor.href = objectURL;
+  anchor.download = filename;
+  anchor.click();
+
+  if (typeof window.URL.revokeObjectURL === 'function') {
```

Comment:
That's an interesting check. Is there a browser or condition in which `window.URL.createObjectURL` exists but `window.URL.revokeObjectURL` doesn't? Looking at caniuse.com they both look to be supported across the browsers we support unless I missed something.

If there is a condition/browser in which this doesn't exist, I would add a comment above this line of when we can remove the typeof check.

Was curious if there is a problem with the `revokeObjectURL` with large blobs and invoking it too soon. I found some commentary here:
https://stackoverflow.com/questions/19327749/javascript-blob-filename-without-link

<img width="759" alt="Screen Shot 2021-06-04 at 9 36 48 AM" src="https://user-images.githubusercontent.com/1151048/120827189-69cf1c80-c518-11eb-95dd-fe87bd32ca74.png">

I don't know if that is going to be problem these days and if you have to schedule this for a later time to be super safe or not. Curious if you tried large files and exports with lots and lots of data to ensure this call will be ok?

Reply:
I carried over the `revokeObjectURL ` logic from the `GenericDownloader` component (https://github.com/elastic/kibana/pull/100554/files#diff-a3d882482407c7bd482ae952442eafe9c68a8a465215cea8b53148af28468ad6L78) without any changes. So I guess it should be working fine. But I haven't tested it on big files tbh.

Reply:
I just tried downloading 8192 rules (~8Mb of data) in Safari, Chrome, FF. Everything works fine. So I guess scheduling `revokeObjectURL` for a later time is something from the dark ages of the Web.